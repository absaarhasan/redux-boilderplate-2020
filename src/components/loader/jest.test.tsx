import React from 'react';
import { cleanup, screen } from '@testing-library/react';
import { render } from '../../../jest.setup';
import '@testing-library/jest-dom';
import Loader from '.';

afterEach(cleanup);

describe('<Loader /> spec', () => {
  it('should be hidden by default', () => {
    render(<Loader />);
    expect(screen.queryByTestId('loader-container')).toBeNull();
  });
  it('should display when triggered', () => {
    const initialState = {
      app: {
        isLoading: true,
      },
    };
    render(<Loader />, initialState);
    expect(screen.getByTestId('loader-container')).toMatchSnapshot();
  });
});
