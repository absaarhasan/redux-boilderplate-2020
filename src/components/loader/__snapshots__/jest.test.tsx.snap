// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Loader /> spec should display when triggered 1`] = `
<div
  class="css-i1mpon-Loader"
  data-testid="loader-container"
>
  <p>
    Loading ...
  </p>
</div>
`;
